src/react/FlowConfigurationPage.js

```diff
@@ -16,136 +16,131 @@
    limitations under the License.
 */
 
-import React from "react";
+import React, { useCallback, useMemo, useState } from "react";
 import { Button, Form, Spinner } from "react-bootstrap";
 
 import SingleCardPage from "./ui/SingleCardPage.js";
 
-export default class FlowConfigurationPage extends React.Component {
-  constructor(props) {
-    super(props);
+let alertId = 0;
 
-    this.astarte = this.props.astarte;
-
-    this.addAlert = this.addAlert.bind(this);
-    this.closeAlert = this.closeAlert.bind(this);
-    this.onFlowNameChange = this.onFlowNameChange.bind(this);
-    this.onConfigChange = this.onConfigChange.bind(this);
-    this.createFlow = this.createFlow.bind(this);
-
-    this.state = {
-      alerts: new Map(),
-      alertId: 0,
-      flowName: "",
-      config: "{}",
-      parsedConfig: null
-    }
-  }
-
-  addAlert(message) {
-    this.setState((state) => {
-      const newAlertId = state.alertId + 1;
-      let newAlerts = state.alerts;
-      newAlerts.set(newAlertId, message);
-
-      return Object.assign(state, {
-        alertId: newAlertId,
-        alerts: newAlerts
-      });
-    });
-  }
-
-  closeAlert(alertId) {
-    this.setState((state) => {
-      state.alerts.delete(alertId);
-      return state;
-    });
-  }
-
-  onFlowNameChange(e) {
-    const newValue = e.target.value;
-
-    this.setState({
-      flowName: newValue
-    });
-  }
-
-  onConfigChange(e) {
-    const newValue = e.target.value;
-    let parsedConfig;
+export default ({ astarte, history, pipelineId }) => {
+  const [alerts, setAlerts] = useState(new Map());
+  const [flow, setFlow] = useState({
+    name: "",
+    config: "{}",
+  });
+  const [isCreatingFlow, setIsCreatingFlow] = useState(false);
 
+  const parsedFlowConfig = useMemo(() => {
     try {
-      parsedConfig = JSON.parse(newValue);
-    } catch (e) {
-      parsedConfig = null;
+      return JSON.parse(flow.config);
+    } catch {
+      return null;
     }
+  }, [flow.config]);
```

Comment:
I'm not sure this works as intended.
check https://github.com/facebook/react/issues/15924

Due to how objects are copied and checked it may be never updated or updated whenever `flow` changes.

Reply:
I got confused here, flow.config is a string, not an object.
The string may represent a valid object, but it's still a string so it will be compared by value